test/test_ops.py

```diff
@@ -340,6 +340,73 @@ def test_out(self, device, dtype, op):
         op(*sample.input, *sample.args, **out_kwargs)
         self.assertEqual(expected, out)
 
+    @ops(op_db)
+    def test_out_empty_tensor(self, device, dtype, op):
+        if not op.supports_tensor_out:
+            self.skipTest("Skipped! Operator %s does not support out=..." % op.name)
+
+        samples = op.sample_inputs(device, dtype)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        # NOTE: only tests on first sample
+        sample = samples[0]
+        # call it normally to get the expected result
+        expected = op(*sample.input, *sample.args, **sample.kwargs)
+        # call it with out=... and check we get the expected result
+        out_kwargs = sample.kwargs.copy()
+        out_kwargs['out'] = out = torch.empty(0,0)
```

Comment:
This torch.empty won't always work because it may not have a dtype that is safe-castable to or be on the input's device. For example, if the input is CUDA and complex64 then this out tensor will be on the CPU and in float32, and that's an invalid out tensor for most ops when given that input.